proposals/lip-introduce_a_genric_keystore.md

```diff
@@ -0,0 +1,276 @@
+
+### Encryption File Format
+
+We store secret recovery phrases and private keys in a JSON file following the format below.
+
+```java
+keystoreSchema = {
+  "type": "object",
+  "required": ["encryptedPassphrase", "metadata", "ID"],
+  "properties": {
+    "encryptedPassphrase": {
+      "type": "object",
+      "properties": {
+        "version": {"type": "string"},
+        "ciphertext": {"type": "string"},
+        "mac": {"type": "string"},
+        "kdf": {"type": "string"},
+        "kdfparams": {"type": "object"},
+        "cipher": {"type": "string"},
+        "cipherparams": {"type": "object"}
+      }
+    },
+    "metadata": {
+        "name": {"type": "string"},
+        "description": {"type": "string"},
+        "pubkey": {"type": "string"},
+        "address": {"type": "string"},
+        "path": {"type": "string"},
+        "derivedFromID": {"type": "string"},
+        "creationTime": {"type": "string"},
+        "pathsUsed": {"type": "array"},
+        "tags": {"type": "array"}
+    },
+    "ID": {
+        "type": "string",
+        "format": "uuid"
+    }
+  }
+}
+```
+
+In the following sections, we describe the uses of the properties of `keystoreSchema`.
+
+#### encryptedPassphrase
+
+##### version
+
+The `version` is set to `"1"`.
+
+##### ciphertext
+
+The encrypted message in hexadecimal format.
+
+##### mac
+
+Computed as `SHA256(encryptionKey[-16:] + cipherText)`, where `encryptionKey` is the key derived from the user password using the key-derivation function. It can be used to verify the encryption key before starting the decryption process.
+
+##### kdf and kdfparams
+
+The encryption/decryption key is an intermediate key derived from the user password. It is used to generate the secret key for decryption, and verify if the given password is correct. The function, and the params used to derive this key from the password are specified in `kdf`. The following values of `kdf` and `kdfparams` are allowed, depending on the key-derivation function:
+
+| `kdf` | function | `kdfparams` | Definition |
+|-------|----------|-------------|------------|
+| "PBKDF2-SHA-256" | `pbkdf2`    | `{iterations: uint32, salt: string}` | [RFC 2898](https://www.ietf.org/rfc/rfc2898.txt)|
+| "argon2id"       | `argon2id`  | `{parallelism: uint32, iterations: uint32, memory: uint32, salt: string}` | [RFC 9106](https://datatracker.ietf.org/doc/html/rfc9106)|
+
+##### cipher and cipherparams
+
+The specified function encrypts the secret using the decryption key; to decrypt it, the decryption key along with `cipher` and `cipherparams` must be used. If the decryption key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
```

Comment:
```suggestion
The specified function encrypts the secret key using the encryption key; to decrypt it, the decryption key along with `cipher` and `cipherparams` must be used. If the decryption key is longer than the key size required by the encoding function, it is truncated to the correct number of bits. The following option is supported:
```

Reply:
I updated several instances in the LIP, to clarify that there is indeed only on key